src/main/java/io/leonis/subra/ipc/network/GameStatePublisher.java

```diff
@@ -21,48 +25,49 @@
 @AllArgsConstructor
 public final class GameStatePublisher implements Publisher<GameState> {
   private final Publisher<WrapperPacket> visionPublisher;
-  private final Publisher<org.robocup.ssl.RefereeOuterClass.Referee> refboxPublisher;
+  private final Publisher<SSL_Referee> refboxPublisher;
 
   @Override
   public void subscribe(final Subscriber<? super GameState> subscriber) {
     Flux.combineLatest(
-        Flux.from(this.visionPublisher)
-            .transform(new SSLVisionDeducer()),
-        Flux.from(this.refboxPublisher)
-            .transform(new SSLRefboxDeducer()),
-        GameState::build)
-        .subscribe(subscriber);
+        Flux.from(this.visionPublisher).transform(new SSLVisionDeducer()),
+        Flux.from(this.refboxPublisher).transform(new SSLRefboxDeducer()),
+        GameStateWithoutGoals::new)
+      .transform(new ParallelDeducer<>(
+          new IdentityDeducer<>(),
+          new GoalsDeducer<>(),
+          GameState::new
+      ))
+      .subscribe(subscriber);
   }
 
   @Value
-  @AllArgsConstructor
-  public static class GameState
-      implements Player.SetSupplier, Goal.SetSupplier, Field.Supplier, Ball.SetSupplier,
-      Referee.Supplier, Temporal {
+  static class GameStateWithoutGoals
```

Comment:
should this be public?

Reply:
I think it migth even be private, since this state won't every leave the `GameStatePublisher`.